lib/lines.js

```diff
@@ -448,6 +448,22 @@ Lp.guessTabWidth = function() {
     return secret.cachedTabWidth = result;
 };
 
+// Determine if the list of lines has a first line that's empty of non-comment
+// tokens. If this is the case, the code may need to be wrapped in parens to
+// avoid ASI issues. Calling code should be more defensive if we return true,
+// so false positives are ok.
+Lp.isFirstLineOnlyComment = function () {
+    var secret = getSecret(this);
+    if (secret.infos.length === 0) {
+        return false;
+    }
+    var firstLineInfo = secret.infos[0],
+        firstLine = firstLineInfo.line.trim();
```

Comment:
> The `secret.infos[i].line` string may have been inherited from another `Lines` object, so it needs to be sliced from `secret.infos[i].sliceStart` to `secret.infos[i].sliceEnd`, then trimmed.

Thanks, I'll update the PR to fix that.

> What if `firstLine.startsWith("/*")` but then later has a `*/` followed by non-comment tokens?

Then it would be a false positive and wrapped in parens when it technically doesn't need to be, which is still correct. (See the comment on the function for a little more of the rationale.) It doesn't seem like there's a great way to detect this case with just string matching (while also returning true for cases like `/* a */ /*`), and doing real lexing seems like needless complexity, but happy to hear suggestions on ways to improve the check.